dev/background-api.py

```diff
@@ -0,0 +1,277 @@
+from root_numpy import random_sample
+from ROOT import TF1
+
+alt = Angle(90 - 30, "deg")
+az = Angle(200, "deg")
+time = Quantity(20, "h")
+
+##bg_event_list = simulate_toy_bg_one_bin(alt, az, time)
+
+# bg model:
+# - spatial coordinates: Gaussian w.r.t. center of FoV
+#   TODO: shouldn't this depend on alt, az?
+#   desirable: use radial aceptance curves, depending on alt, az.
+# - energy: bg spectrum multiplied by the effective area
+# for the bg spectrum one can use the Hegra spectrum:
+#  power-law with index 2.7
+#  ref: http://journals.aps.org/prd/pdf/10.1103/PhysRevD.59.092003
+# alternative: use BESS spectrum for low-E cut-off:
+# http://iopscience.iop.org/0004-637X/545/2/1135
+width = Angle(2.0, "deg")  # for instance, no idea what it should be...
+bg_model_spatial = Gaussian(width)
+E_0 = Quantity(1, "TeV")
+bg_spectrum = NORM_E * pow(E / E_0, 2.7)  # take NORM_E from Hegra paper
+# does this exist?
+Aeff = find_lookup('effective_area', altitude='alt', azimuth='az')
+bg_model_energy = bg_spectrum * Aeff
+bg_model = bg_model_spatial * bg_model_energy
+
+# generate bg
+n_events = quad(bg_model_energy * Aeff) * time  # integrate in energy
+
+##DETX, DETY = random(bg_model_spatial, n_events)
+##energy = random(bg_model_energy, n_events)
+
+offset = random_sample(TF1("f1", bg_model_spatial), n_events, seed=1)
+angle = random_sample(TF1("f1", "2*TMath::Pi()"), n_events, seed=1)
+
+DETX, DETY = transform_polar_to_cartesian(offset, angle)
+
+energy = random_sample(TF1("f1", bg_model_energy), n_events, seed=1)
+
+# refs for random generation:
+# http://rootpy.github.io/root_numpy/reference/generated/root_numpy.random_sample.html
+# http://www.anderswallin.net/2009/05/uniform-random-points-in-a-circle-using-polar-coordinates/
+# https://docs.python.org/2/library/random.html
+
+# produce event list
+bg_event_list = EventList(DETX, DETY, energy)
+# TODO: store header information (alt, az, time)
+
+
+# Use case: implement FoV bg method
+# - given a list of ON runs
+# - subtract the bg according to the FoV method from Berge 2007 in 2D (X, Y)
+#   or 3D (X, Y, E) using existing bg models
+# - deliver absolute numbers (bg statistics) and maps or cubes
+
+
+# Use case: implement ring bg method
+
+
+# Use case: implement ON/OFF bg method
+# - given a list of ON runs and a list of OFF runs
+# - subtract the bg from the ON runs according to the ON/OFF method from Berge
+#   2007 in 2D (X, Y) or 3D (X, Y, E) developping bg models from the OFF runs
+# - deliver absolute numbers (bg statistics) and maps or cubes
+
+
+# Use case (high-level): GC maps with ON/OFF bg
+# - make a significance and flux image of the Galactic center for HESS with
+#   on-off background method.
+# - Apply on-off background method, i.e. given on- and off-run list, produce
+#   images or cubes for n_on, n_off, exp_on, exp_off.
+
+
+# Use case: compute background IRFs in CTA format from HESS off runs
+# ref: https://github.com/gammapy/gammapy/issues/267
```

Comment:
Is this different from the first use case listed above?
Or is there just one last step or extra class that collects the background rates into a single container and stores them in FITS?